### ISSUE TITLE
Add Automatic-Module-Name to JAR manifest for better Java Module System behavior

### ISSUE THREAD

**Issue by msgilligan** [JavaMoney/jsr354-api-bp#10]
Currently the module names seem to be:

* `jsr354-api-bp`: `javax.money`
* `jsr354-ri-bp`: `moneta.bp`

For the the non-backported versions they are:

* `jsr354-api`: `java.money`
* `jsr354-ri`: `org.javamoney.moneta`

I'm not sure if the `-bp` versions and the non-bp versions should have the same module name or not (there might be some issues if they're the same), but at the very least the backport version should have a module name starting with `org.javamoney.`. 

It would also be nice to see some (more?) documentation on "the big picture" for how JavaMoney is handling Java Modules in both the backport and non-backport versions.

p.s. I'm willing to help with this.

**Comment by keilw** [comment]
There is no need for Java 9 modules in the Backports. They are entirely for a Java 6 or 7 legacy environments, once you have at least Java 8 the new versions are heavily recommended. Hence we don't plan much but you are welcome to propose a PR for the backports, we could release one or the other version of the BP to get the automatic names in sync (the API should also say java.money because you are not supposed to use both in the same module path ;-)

**Comment by msgilligan** [comment]
I'm building a suite of applications (JavaFX, Android, command-line, and web-services) that use JavaMoney. I have a handful of component JARs containing various functionality.

The place where I am using modules is in JavaFX applications. I'm building them using JavaFX 14, Java 14, `package`, and the [Badass JLink Plugin](https://github.com/beryx/badass-jlink-plugin) to build bundled "native" applications.

Since I'm thinking about making Android 9 the minimum Android I support, I decided on monday to try upgrading from the JavaMoney BP dependencies (api bp and ri bp) to the latest non-BP dependencies. Because all my JavaFX-related libraries are modular I had to change the `module-info.java` files when I switched to non-BP. I then ran into problems with the modular JavaMoney (the problems may be specific to the processing done by the Badass JLink Plugin) and decided to switch back to the BP versions. I think had to change all the `module-info.java` files back to the BP module names.

So, in complex, real-world ecosystems it remains necessary to support both modular and non-modular apps. With Android in the mix, I think the BP jars should play as nicely as possible with the non-BP jars. Using logical (but perhaps not identical, if that is necessary) automatic-module-names would be helpful. It would also be nice to see some documentation on the modular architecture of JavaMoney is supposed to work.  (For example, what do I need to do to make the Bitcoin currency provider workin in both modular and non-modular environments?)
